src/passes/SimplifyLocals.cpp

```diff
@@ -147,6 +146,9 @@ struct SimplifyLocals
         self->unoptimizableBlocks.insert(target);
       }
       // TODO: we could use this info to stop gathering data on these blocks
+    } else if (auto* br = curr->dynCast<BrOnExn>()) {
+      // We cannot sink br_on_exn out of its targeting block
```

Comment:
minor note on the text here: i believe the issue is not that we can't sink a `br_on_exn`, but that if there is one, it inhibits optimizing a block return value for the block it targets. "sink" is a (not very good) term used for moving a `set` to the corresponding `get`.

so the text might be `We cannot optimize the block this targets to have a return value, as the `br_on_exn` doesn't support a change to the block's type` or something like that.

Reply:
Done